### PR TITLE
【Deepmd Support】add IsInitialized and tanh double grad

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

New features
### PR changes

OPs
### Describe

This PR add IsInitialized API into paddle::Tensor to support user check whether their input tensor is ready to support deepmd custom op in PR [https://github.com/deepmodeling/deepmd-kit/pull/499](url)
And Add tanhDoubleGrad kernel to support  SEA net work of deepmd
[https://github.com/deepmodeling/deepmd-kit/blob/api/examples/water/train/water_se_a.json](url)

### CODE REVIEW

File: paddle/fluid/extension/include/ext_tensor.h [hunk 1]
```diff
@@ -113,6 +113,9 @@ class PD_DLL_DECL Tensor {
   /// \brief Cast datatype from one to another
   Tensor cast(const DataType& target_type) const;
 
+  /// \brief Check Tensor is initialized
+  bool IsInitialized() const;
```
Comment (chenwhql) [inline paddle/fluid/extension/include/ext_tensor.h]: API名其他都是小写，都统一成小写？

**Review by ceci3 — COMMENTED** [review]
tanh double grad LGTM

File: paddle/fluid/extension/include/ext_tensor.h (hunk 1 above)
Comment (JiabinYang) [reply paddle/fluid/extension/include/ext_tensor.h]: sure

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**